shared/metrics/src/main/java/io/pravega/shared/metrics/DynamicLoggerImpl.java

```diff
@@ -95,6 +95,24 @@ public Counter call() throws Exception {
     }
 
     @Override
+    public long getCounterValue(String name) {
+        Exceptions.checkNotNullOrEmpty(name, "name");
+        String counterName = name + ".Counter";
+        try {
+            Counter counter = countersCache.get(counterName, new Callable<Counter>() {
+                @Override
+                public Counter call() throws Exception {
+                    return underlying.createCounter(counterName);
```

Comment:
this is a loading cache, in case the value is not present in cache, it uses the supplied loader to load the value.. 
the problem here, though, is that loader is simply creating a new counter which will start with count 0. 
in another comment i had asked abhijit to ensure that a lambda be passed which will be used to compute the correct value and load it in cache.